src/App.jsx

```diff
@@ -14,57 +16,73 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  const selectGood = good => () => {
+    setSelectedGood(good);
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+  const unSelectGood = () => () => {
+    setSelectedGood('');
+  };
```

Comment:
The `unSelectGood` function is unnecessarily wrapped in another function. It should be defined as a simple function without returning another function. This will simplify the code and make it more readable. You can define it as `const unSelectGood = () => { setSelectedGood(''); };` and use it directly in the `onClick` handlers.

Reply:
agree

Reply:
this is the callback for onClick:
```
const unSelectGood = () => () => { /.../ }
<button onClick={unSelectGood()} />
```
it's the same as
```
const unSelectGood = () => { /.../ }
<button onClick={() => unSelectGood()} />
```
I changed it